### PR TITLE
CANS-1053 Print disabled until age template is selected

### PR DESCRIPTION
**Pull request by nasar-cambria**
https://osi-cwds.atlassian.net/browse/CANS-1053

## Description
To Disable Print button until age is button is selected and hide discard and continue button until assessment is saved. 

## Tests

- [ ] I used TDD to develop the feature
- [ ] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/Application/components/Header/PageHeaderButtons/PrintButton.js [hunk 1]
```diff
@@ -52,9 +52,10 @@ class PrintButton extends Component {
   }
 
   renderPrintButton = () => {
-    const { isEnabled } = this.props
+    const { isEnabled, isUnderSix, isValidDate } = this.props
+    const isPrintButtonEnabled = isUnderSix !== undefined && isValidDate ? isEnabled : !isEnabled
```
Comment (wmitchell-cambria) [inline app/javascript/Application/components/Header/PageHeaderButtons/PrintButton.js]: If it is not enabled, but you select an invalid date, then it is...enabled?

File: app/javascript/Application/components/Assessment/AssessmentContainer.js [hunk 2]
```diff
@@ -109,7 +109,15 @@ export default class AssessmentContainer extends Component {
     const { assessment, i18n, isEditable } = this.state
     const node = <PrintAssessment assessment={assessment} i18n={i18n} />
     const leftButton = isEditable ? buildSaveAssessmentButton(this.handleSaveAssessment, isSaveButtonEnabled) : null
-    const rightButton = <PrintButton node={node} isEnabled={true} isAssessmentRendered={true} />
+    const rightButton = (
+      <PrintButton
+        node={node}
+        isEnabled={true}
```
Comment (wmitchell-cambria) [inline app/javascript/Application/components/Assessment/AssessmentContainer.js]: It seems like we could just use the `isEnabled` prop here, and calculate whether or not it is enabled using the `under_six` and `isValidDate` info. Why add more props to the Print button when we already have one that does exactly what we want?

Also, remember that Print is independent of Assessment. It is also used on the change log page. `isAssessmentRendered`, `isUnderSix`, and `isValidDate` do not make sense in that context.

Comment (nasar-cambria) [reply app/javascript/Application/components/Assessment/AssessmentContainer.js]: I Agree. Thanks!

File: app/javascript/Application/components/Header/PageHeaderButtons/PrintButton.js (hunk 1 above)
Comment (nasar-cambria) [reply app/javascript/Application/components/Header/PageHeaderButtons/PrintButton.js]: It works just like save button. The AC only says when date is not available it should be disabled.

**Comment by nasar-cambria** [comment]
@wmitchell-cambria when you get a chance can you please revisit ?

**Comment by wmitchell-cambria** [comment]
Just waiting on a unit test or two for this one.

**Comment by LeonidMarushevskyi** [comment]
Retest this please

**Comment by nasar-cambria** [comment]
Jenkins Re-test this please.

**Comment by LeonidMarushevskyi** [comment]
Jenkins Re-test this please.

**Comment by nasar-cambria** [comment]
jenkins re-test this please

**Pull request merged**